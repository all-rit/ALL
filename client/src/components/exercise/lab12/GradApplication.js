/* eslint-disable no-unused-vars */
// import useMainStateContext from "src/reducers/MainContext";
import React, { useState } from "react";

const Application = () => {
  const [firstName, setFirstName] = useState("");
  const [lastName, setLastName] = useState("");
  const [pronouns, setPronouns] = useState("");
  const [college, setCollege] = useState("");
  const [major, setMajor] = useState("");
  const [gradTerm, setGradTerm] = useState("");

  const [fNameErr, setFirstNameErr] = useState(false);
  const [lNameErr, setLastNameErr] = useState(false);
  const [pronounsErr, setPronounsErr] = useState(false);
  const [collegeErr, setCollegeErr] = useState(false);
  const [majorErr, setMajorErr] = useState(false);
  const [gradTermErr, setGradTermErr] = useState(false);

  const [fNameEmptyErr, setFirstNameEmptyErr] = useState(false);
  const [lNameEmptyErr, setLastNameEmptyErr] = useState(false);
  const [pronounsEmptyErr, setPronounsEmptyErr] = useState(false);
  const [collegeEmptyErr, setCollegeEmptyErr] = useState(false);
  const [majorEmptyErr, setMajorEmptyErr] = useState(false);
  const [gradTermEmptyErr, setGradTermEmptyErr] = useState(false);

  const validateInput = () => {
    let error = false;
    setFirstNameEmptyErr(false);
    setFirstNameErr(false);
    setLastNameEmptyErr(false);
    setLastNameErr(false);
    setPronounsEmptyErr(false);
    setPronounsErr(false);
    setCollegeEmptyErr(false);
    setCollegeErr(false);
    setMajorEmptyErr(false);
    setMajorErr(false);
    setGradTermEmptyErr(false);
    setGradTermErr(false);

    // RegExp catches on anything but white space
    const hasCharacter = new RegExp("[\\S]");
    const hasLetter = new RegExp("[A-Za-z]");

    // RegExp for special characters (anything except letters and whitespace));
    const fnSpecialChar = new RegExp("[^A-Za-z\\s+]", "g");
    if (fnSpecialChar.test(firstName)) {
      error = true;
      setFirstNameErr(true);
    }
    if (!hasCharacter.test(firstName)) {
      error = true;
      setFirstNameEmptyErr(true);
    }

    // RegExp for special characters (anything except letters and whitespace));
    const lnSpecialChar = new RegExp("[^A-Za-z\\s+]", "g");
    if (lnSpecialChar.test(lastName)) {
      error = true;
      setLastNameErr(true);
    }
    if (!hasCharacter.test(lastName)) {
      error = true;
      setLastNameEmptyErr(true);
    }

    const pReg = new RegExp("[^A-Za-z\\s+/]", "g");
    if (pReg.test(pronouns)) {
      error = true;
      setPronounsErr(true);
    }
    if (!hasCharacter.test(pronouns)) {
      error = true;
      setPronounsEmptyErr(true);
    }

    const cReg = new RegExp("[^A-Za-z\\s+.]", "g");
    if (cReg.test(college)) {
      error = true;
      setCollegeErr(true);
    }
    if (!hasCharacter.test(college) || !hasLetter.test(college)) {
      error = true;
      setCollegeEmptyErr(true);
    }

    const mReg = new RegExp("[^A-Za-z\\s+.]", "g");
    if (mReg.test(major)) {
      error = true;
      setMajorErr(true);
    }
    if (!hasCharacter.test(major) || !hasLetter.test(major)) {
      error = true;
      setMajorEmptyErr(true);
    }

    const gtReg = new RegExp("[^A-Za-z0-9\\s+]", "g");
    if (gtReg.test(gradTerm)) {
      error = true;
      setGradTermErr(true);
    }
    if (!hasCharacter.test(gradTerm)) {
      error = true;
      setGradTermEmptyErr(true);
    }
    // -------------------------------------------------
  };
  return (
    <div>
      <div style={{ display: "flex", justifyContent: "center" }}>
        <div className="tw-mt-10 tw-shadow-2xl-top-bottom tw-rounded-3xl tw-min-w-72 tw-w-9/12 lg:tw-w-5/12 md:tw-w-7/12 tw-h-full tw-p-10">
          <h1 className="tw-text-3xl tw-font-bold tw-mb-8">
            Apply for Graduation at ALL University
          </h1>
          <div className="tw-text-left tw-p-6">
            <h2 className="tw-text-xl tw-font-semibold tw-mb-1.5">
              Personal Information:{" "}
            </h2>
            <div className="tw-resize-x tw-flex1 tw-mb-4">
              <label className="tw-pr-8">Legal First Name:</label>
              <input
                placeholder="Ex: Jane" className="tw-h-8 tw-w-full md:tw-w-6/12"
                onChange={(e) => {
                  setFirstName(e.target.value);
                }}
              />
              <br />
              {fNameErr && (
                <label className="tw-text-error-red tw-text-sm tw-pl-4 tw-italic">
                  Error: Invalid character used.
                </label>
              )}
              {fNameEmptyErr && (
                <label className="tw-text-error-red tw-text-sm tw-pl-4 tw-italic">
                  Error: Input required.
                </label>
              )}
            </div>
            <div className="tw-flex1 tw-mb-4">
              <label className="tw-pr-8">Legal Last Name:</label>
              <input
                placeholder="Ex: Smith" className="tw-h-8 tw-w-full md:tw-w-6/12"
                type="text"
                onChange={(e) => {
                  setLastName(e.target.value);
                }}
              />
              <br />
              {lNameErr && (
                <label className="tw-text-error-red tw-text-sm tw-pl-4 tw-italic">
                  Error: Invalid character used.
                </label>
              )}
              {lNameEmptyErr && (
                <label className="tw-text-error-red tw-text-sm tw-pl-4 tw-italic">
                  Error: Input required.
                </label>
              )}
            </div>
            <div className="tw-flex1 tw-mb-4">
              <label className="tw-pr-8">Pronouns:</label>
              <input
                placeholder="Ex: They/Them" className="tw-h-8 tw-w-full md:tw-w-6/12"
                type="text"
                onChange={(e) => {
                  setPronouns(e.target.value);
                }}
              />
              <br />
              {pronounsErr && (
                <label className="tw-text-error-red tw-text-sm tw-pl-4 tw-italic">
                  Error: Invalid character used.
                </label>
              )}
              {pronounsEmptyErr && (
                <label className="tw-text-error-red tw-text-sm tw-pl-4 tw-italic">
                  Error: Input required.
                </label>
              )}
            </div>
            <h3 className="tw-mt-10 tw-text-xl tw-font-semibold tw-mb-1.5">
              Academic Information:{" "}
            </h3>
            <div className="tw-flex1 tw-mb-4">
              <label className="tw-pr-8">College:</label>
              <input
                placeholder="Ex: RIT" className="tw-h-8 tw-w-full md:tw-w-6/12"
                type="text"
                onChange={(e) => {
                  setCollege(e.target.value);
                }}
              />
              <br />
              {collegeErr && (
                <label className="tw-text-error-red tw-text-sm tw-pl-4 tw-italic">
                  Error: Invalid character used.
                </label>
              )}
              {collegeEmptyErr && (
                <label className="tw-text-error-red tw-text-sm tw-pl-4 tw-italic">
                  Error: Input required.
                </label>
              )}
            </div>
            <div className="tw-flex1 tw-mb-4">
              <label className="tw-pr-8">Major:</label>
              <input
                placeholder="Ex: CS" className="tw-h-8 tw-w-full md:tw-w-6/12"
                type="text"
                onChange={(e) => {
                  setMajor(e.target.value);
                }}
              />
              <br />
              {majorErr && (
                <label className="tw-text-error-red tw-text-sm tw-pl-4 tw-italic">
                  Error: Invalid character used.
                </label>
              )}
              {majorEmptyErr && (
                <label className="tw-text-error-red tw-text-sm tw-pl-4 tw-italic">
                  Error: Input required.
                </label>
              )}
            </div>
            <div className="tw-flex1 tw-mb-4">
              <label className="tw-pr-8">Graduation Term:</label>
              <input
                placeholder="Ex: Spring 2024" className="tw-h-8 tw-w-full md:tw-w-6/12"
                type="text"
                onChange={(e) => {
                  setGradTerm(e.target.value);
                }}
              />
              <br />
              {gradTermErr && (
                <label className="tw-text-error-red tw-text-sm tw-pl-4 tw-italic">
                  Error: Invalid character used.
                </label>
              )}
              {gradTermEmptyErr && (
                <label className="tw-text-error-red tw-text-sm tw-pl-4 tw-italic">
                  Error: Input required.
                </label>
              )}
            </div>
          </div>
          <button
            className="tw-text-error-red tw-mt-8 btn-primary btn btn-md"
            onClick={() => {
              validateInput();
            }}
          >
            Submit Application
          </button>{" "}
        </div>
      </div>
    </div>
  );
};

export default Application;
