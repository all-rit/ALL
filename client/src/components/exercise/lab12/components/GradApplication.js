import React, { useState, useEffect, useContext } from "react";
import useMainStateContext from "src/reducers/MainContext";
import ExerciseStateContext from "../Lab12Context";
import { navigate } from "@reach/router";
import { ExerciseService } from "../../../../services/lab12/ExerciseService";
import {
  Button,
  ButtonDropdown,
  DropdownItem,
  DropdownMenu,
  DropdownToggle,
  Form,
  FormGroup,
  Input,
  Label,
} from "reactstrap";

const GradApplication = () => {
  const { state } = useMainStateContext();
  const user = state.main.user;

  const [isRepairComplete, setIsRepairComplete] = useState(false);
  const fetchExercise = async () => {
    try {
      const currentExercise = await ExerciseService.fetchExercise({
        userid: user.userid,
      });
      setIsRepairComplete(currentExercise.isFormRepairComplete);
    } catch (error) {
      console.error("Could not fetch exercise: ", error);
    }
  };

  const [dropdownOpen, setDropdownOpen] = useState(false);
  const [dropdownLabel, setDropdownLabel] = useState("Select Pronouns");

  const setPronounsAndDropdownLabel = (pronouns) => {
    setPronouns(pronouns);
    setDropdownLabel(pronouns);
  };

  useEffect(() => {
    fetchExercise();
  }, []);

  const {
    firstName,
    setFirstName,
    lastName,
    setLastName,
    preferredName,
    setPreferredName,
    pronouns,
    setPronouns,
    college,
    setCollege,
    major,
    setMajor,
    gradTerm,
    setGradTerm,
  } = useContext(ExerciseStateContext);

  const [formErrors, setFormErrors] = useState([]);

  const validateInput = (formData) => {
    const errors = formData.map((input) => input === "");
    const hasError = errors.some((error) => error);
    setFormErrors(errors);
    return hasError;
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    const formData = !isRepairComplete
      ? [firstName, lastName, college, major, gradTerm]
      : [
          firstName,
          lastName,
          preferredName,
          pronouns,
          college,
          major,
          gradTerm,
        ];

    const error = validateInput(formData); // Validate the form

    if (!error) {
      navigate(`/Lab12/Exercise/PreWrongDiploma`); // If no errors, submit
    } else {
      console.log("Form contains errors.");
    }
  };

  return (
    <Form onSubmit={handleSubmit}>
      <div className="tw-flex tw-justify-center">
        <div className="tw-mt-10 tw-shadow-2xl-top-bottom tw-rounded-3xl tw-min-w-72 tw-w-9/12 lg:tw-w-7/12 tw-h-full tw-p-10">
          <h1 className="tw-text-3xl tw-font-bold tw-mb-8">
            Apply for Graduation at ALL University
          </h1>
          <div className="tw-text-left tw-p-6">
            <h2 className="tw-text-xl tw-font-semibold tw-mb-1.5">
              Personal Information:{" "}
            </h2>
            <FormGroup className="sm:tw-flex tw-items-center tw-mb-6">
              <Label htmlFor="firstName" className="tw-pr-8 tw-mb-0">
                <span className="tw-text-error-red">*</span>
                Legal First Name:
              </Label>
              <div className="tw-flex tw-flex-col tw-max-w-72 tw-w-full sm:tw-w-8/12 md:tw-w-6/12">
                <Input
                  type="text"
                  id="firstName"
                  label="Legal First Name"
                  placeholder="Ex: Jane"
                  value={firstName}
                  onChange={(e) => {
                    setFirstName(e.target.value);
                  }}
                />
                {formErrors[0] && (
                  <Label
                    htmlFor="firstName"
                    className="tw-text-error-red tw-text-sm tw-pl-4 tw-italic"
                    data-testid="invalid-char"
                  >
                    Error: Invalid character.
                  </Label>
                )}
              </div>
            </FormGroup>
            <FormGroup className="sm:tw-flex tw-items-center tw-mb-6">
              <Label htmlFor="lastName" className="tw-pr-8 tw-mb-0">
                <span className="tw-text-error-red">*</span>
                Legal Last Name:
              </Label>
              <div className="tw-flex tw-flex-col tw-max-w-72 tw-w-full sm:tw-w-8/12 md:tw-w-6/12">
                <Input
                  type="text"
                  id="lastName"
                  placeholder="Ex: Smith"
                  value={lastName}
                  onChange={(e) => {
                    setLastName(e.target.value);
                  }}
                />
                {formErrors[1] && (
                  <Label
                    htmlFor="lastName"
                    className="tw-text-error-red tw-text-sm tw-pl-4 tw-italic"
                  >
                    Error: Input required.
                  </Label>
                )}
              </div>
            </FormGroup>
            {isRepairComplete && (
              <FormGroup className="sm:tw-flex tw-items-center tw-mb-6">
                <Label className="tw-pr-8 tw-mb-0">Preferred Name:</Label>
                <div className="tw-flex tw-flex-col tw-max-w-72 tw-w-full sm:tw-w-8/12 md:tw-w-6/12">
                  <Input
                    type="text"
                    id="preferredName"
                    placeholder="Ex: Jay"
                    value={preferredName}
                    onChange={(e) => {
                      setPreferredName(e.target.value);
                    }}
                  />
                  {formErrors[2] && (
                    <Label
                      htmlFor="preferredName"
                      className="tw-text-error-red tw-text-sm tw-pl-4 tw-italic"
                    >
                      Error: Input required.
                    </Label>
                  )}
                </div>
              </FormGroup>
            )}
            {isRepairComplete && (
              <FormGroup className="sm:tw-flex tw-items-center tw-mb-6">
                <Label className="tw-pr-8 tw-mb-0">
                  <span className="tw-text-error-red">*</span>Pronouns:
                </Label>
                <div className="tw-flex tw-flex-col tw-max-w-72 tw-w-full sm:tw-w-8/12 md:tw-w-6/12">
                  <ButtonDropdown
                    toggle={() => setDropdownOpen(!dropdownOpen)}
                    isOpen={dropdownOpen}
                  >
                    <DropdownToggle color={"secondary"} caret>
                      {dropdownLabel}
                    </DropdownToggle>
                    <DropdownMenu>
                      <DropdownItem
                        onClick={() => setPronounsAndDropdownLabel("He/Him")}
                      >
                        {" "}
                        He/Him{" "}
                      </DropdownItem>
                      <DropdownItem
                        onClick={() => setPronounsAndDropdownLabel("She/Her")}
                      >
                        {" "}
                        She/Her{" "}
                      </DropdownItem>
                      <DropdownItem
                        onClick={() => setPronounsAndDropdownLabel("They/Them")}
                      >
                        {" "}
                        They/Them{" "}
                      </DropdownItem>
                    </DropdownMenu>
                  </ButtonDropdown>
                  {formErrors[3] && (
                    <Label
                      htmlFor="pronouns"
                      className="tw-text-error-red tw-text-sm tw-pl-4 tw-italic"
                    >
                      Error: Please make a selection.
                    </Label>
                  )}
                </div>
              </FormGroup>
            )}
            <h3 className="tw-mt-14 tw-text-xl tw-font-semibold tw-mb-1.5">
              Academic Information:{" "}
            </h3>
            <FormGroup className="sm:tw-flex tw-items-center tw-mb-6">
              <Label className="tw-pr-8 tw-mb-0">College:</Label>
              <div className="tw-flex tw-flex-col tw-max-w-72 tw-w-full sm:tw-w-8/12 md:tw-w-6/12">
                <Input
                  type="text"
                  id="college"
                  placeholder="Ex: RIT"
                  value={college}
                  onChange={(e) => {
                    setCollege(e.target.value);
                  }}
                />
                {isRepairComplete
                  ? formErrors[4]
                  : formErrors[2] && (
                      <Label
                        htmlFor="college"
                        className="tw-text-error-red tw-text-sm tw-pl-4 tw-italic"
                      >
                        Error: Input required.
                      </Label>
                    )}
              </div>
            </FormGroup>
            <FormGroup className="sm:tw-flex tw-items-center tw-mb-6">
              <Label className="tw-pr-8 tw-mb-0">Major:</Label>
              <div className="tw-flex tw-flex-col tw-max-w-72 tw-w-full sm:tw-w-8/12 md:tw-w-6/12">
                <Input
                  type="text"
                  id="major"
                  placeholder="Ex: CS"
                  value={major}
                  onChange={(e) => {
                    setMajor(e.target.value);
                  }}
                />
                {isRepairComplete
                  ? formErrors[6]
                  : formErrors[3] && (
                      <Label
                        htmlFor="major"
                        className="tw-text-error-red tw-text-sm tw-pl-4 tw-italic"
                      >
                        Error: Input required.
                      </Label>
                    )}
              </div>
            </FormGroup>
            <FormGroup className="sm:tw-flex tw-items-center tw-mb-6">
              <Label className="tw-pr-8 tw-mb-0">Graduation Term:</Label>
              <div className="tw-flex tw-flex-col tw-max-w-72 tw-w-full sm:tw-w-8/12 md:tw-w-6/12">
                <Input
                  type="text"
                  id="gradTerm"
                  placeholder="Ex: Spring 2024"
                  value={gradTerm}
                  onChange={(e) => {
                    setGradTerm(e.target.value);
                  }}
                />
                {isRepairComplete
                  ? formErrors[7]
                  : formErrors[4] && (
                      <Label
                        htmlFor="gradTerm"
                        className="tw-text-error-red tw-text-sm tw-pl-4 tw-italic"
                      >
                        Error: Input required.
                      </Label>
                    )}
              </div>
            </FormGroup>
          </div>
          <Button
            type="submit"
            className="tw-text-error-red tw-mt-8 btn-primary btn btn-md"
            onClick={(e) => {
              handleSubmit(e);
            }}
          >
            Submit Application
          </Button>
        </div>
      </div>
    </Form>
  );
};

export default GradApplication;
