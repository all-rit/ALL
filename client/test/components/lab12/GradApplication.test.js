import React from "react";
import renderer from "react-test-renderer";
import GradApplication from "../../../src/components/exercise/lab12/GradApplication";

/*
 * Test that the component has 6 input fileds for
 * LFN, LLN, PRN, COL, MJR, GTM
 * And one "Submit Application" button field
 */
describe("Test GradApplication Component Input Fields", () => {
  it("has 6 input fields", () => {
    const component = renderer.create(<GradApplication />);
    const rootInstance = component.root;

    // Verify header:
    const textElement = rootInstance.findByProps({
      children: "Apply for Graduation at ALL University",
    });
    expect(textElement).toBeTruthy();

    // Verify input fields:
    const inputFields = rootInstance.findAllByType("input");
    expect(inputFields.length).toBe(6);
    // Check placeholder text:
    const firstNameInputPlaceholder = inputFields[0].props.placeholder;
    const lastNameInputPlaceholder = inputFields[1].props.placeholder;
    const pronounsInputPlaceholder = inputFields[2].props.placeholder;
    const collegeInputPlaceholder = inputFields[3].props.placeholder;
    const majorInputPlaceholder = inputFields[4].props.placeholder;
    const gradTermInputPlaceholder = inputFields[5].props.placeholder;
    expect(firstNameInputPlaceholder).toBe("Ex: Jane");
    expect(lastNameInputPlaceholder).toBe("Ex: Smith");
    expect(pronounsInputPlaceholder).toBe("Ex: They/Them");
    expect(collegeInputPlaceholder).toBe("Ex: RIT");
    expect(majorInputPlaceholder).toBe("Ex: CS");
    expect(gradTermInputPlaceholder).toBe("Ex: Spring 2024");

    // Verify button:
    const buttons = rootInstance.findAllByType("button");
    const submitButton = buttons.find(
      (button) => button.props.children === "Submit Application",
    );

    // Check if button with text "Submit Application" exists
    expect(submitButton).toBeTruthy();
  });
});

/*
 * Snapshot testing allows you to automatically save a copy
 * of the rendered output of your components and compare it
 * to the output in future tests. This helps catch
 * unintended changes in your components.
 */

describe("Test GradApplication Component", () => {
  it("renders correctly", () => {
    const component = renderer.create(<GradApplication />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// describe("Test logging", () => {
//   const testRenderer = renderer.create(<GradApplication />);
//   const testInstance = testRenderer.root;

//   // manually trigger the callback
//   renderer.act(() => {
//     tree.props.onMouseEnter();
//   });
//   // re-rendering
//   tree = component.toJSON();
//   expect(tree).toMatchSnapshot();

//   // Trigger an update lifecycle
//   testInstance.findByType('button').props.onClick();
//   // Assert on the new state of the component
//   // expect(testInstance.findByType('button').children[1]).toBe('1');
// });

// describe("Test GradApplication Component", () => {
//   it("renders correctly", () => {
//     const { getByText, getByPlaceholderText } = render(<GradApplication />);

//     // Test if the main heading is rendered ?? is this necessary
//     expect(getByText("Apply for Graduation at ALL University")).toBeInTheDocument();

//     // Test if input fields are rendered
//     expect(getByPlaceholderText("Ex: Jane")).toBeInTheDocument();
//     expect(getByPlaceholderText("Ex: Smith")).toBeInTheDocument();
//     expect(getByPlaceholderText("Ex: They/Them")).toBeInTheDocument();
//     expect(getByPlaceholderText("Ex: RIT")).toBeInTheDocument();
//     expect(getByPlaceholderText("Ex: CS")).toBeInTheDocument();
//     expect(getByPlaceholderText("Ex: Spring 2024")).toBeInTheDocument();
//   });

//   it("validates input correctly", () => {
//     const { getByPlaceholderText, getByText } = render(<GradApplication />);

//     // Fill in input fields with invalid data
//     // fireEvent allows us to simulate user interactions with the application that change its state
//     fireEvent.change(getByPlaceholderText("Ex: Jane"), { target: { value: "Jane123" } });
//     fireEvent.change(getByPlaceholderText("Ex: Smith"), { target: { value: "Smith123" } });
//     fireEvent.change(getByPlaceholderText("Ex: They/Them"), { target: { value: "They/Them123" } });
//     fireEvent.change(getByPlaceholderText("Ex: RIT"), { target: { value: "RIT123" } });
//     fireEvent.change(getByPlaceholderText("Ex: CS"), { target: { value: "CS123" } });
//     fireEvent.change(getByPlaceholderText("Ex: Spring 2024"), { target: { value: "Spring 2024!" } });

//     fireEvent.click(getByText("Submit Application"));

//     // Check if error messages are displayed
//     // idk if this works yet
//     expect(getByText("Error: Invalid character.")).toBeInTheDocument();
//     expect(getByText("Error: Invalid character.")).toBeInTheDocument();
//     expect(getByText("Error: Invalid character.")).toBeInTheDocument();
//     expect(getByText("Error: Invalid character.")).toBeInTheDocument();
//     expect(getByText("Error: Invalid character.")).toBeInTheDocument();
//     expect(getByText("Error: Invalid character.")).toBeInTheDocument();

//     // Fill in input fields with empty data
//     fireEvent.change(getByPlaceholderText("Ex: Jane"), { target: { value: " " } });
//     fireEvent.change(getByPlaceholderText("Ex: Smith"), { target: { value: " " } });
//     fireEvent.change(getByPlaceholderText("Ex: They/Them"), { target: { value: "    " } });
//     fireEvent.change(getByPlaceholderText("Ex: RIT"), { target: { value: ".  " } });
//     fireEvent.change(getByPlaceholderText("Ex: CS"), { target: { value: "" } });
//     fireEvent.change(getByPlaceholderText("Ex: Spring 2024"), { target: { value: "" } });

//     // Check if error messages are displayed
//     expect(getByText("Error: Input required.")).toBeInTheDocument();
//     expect(getByText("Error: Input required.")).toBeInTheDocument();
//     expect(getByText("Error: Input required.")).toBeInTheDocument();
//     expect(getByText("Error: Input required.")).toBeInTheDocument();
//     expect(getByText("Error: Input required.")).toBeInTheDocument();
//     expect(getByText("Error: Input required.")).toBeInTheDocument();
//   });
// });

// // describe("Grad Application Tests", () => {

// //   const render = renderer;

// //   test('Test component has 6 input fields', () => {
// //     const { container } = render(<GradApplication />); // Render component
// //     // Get all input elements within the rendered GradApplication
// //     const inputFields = container.querySelectorAll('input');
// //     // Assert that there are 6 fields (LFN, LLN, PR, COL, MJ, GT)
// //     expect(inputFields.length).toBe(6);
// //   });

// //   test('Test component has 1 button field', () => {
// //     const { container } = render(<GradApplication />);
// //     const buttonField = container.querySelectorAll('button');
// //     expect(buttonField.length).toBe(1);
// //   });

// //   test("LFN", async () => {
// //     const user = userEvent.setup();

// //     const updateCallback = vi.fn()
// //     render(<MyTest value="" updateCallback={updateCallback}/>)

// //     const input = screen.getByLabelText("My Test");

// //     await act(async () => await user.type(input, "my-value"));

// //     expect(updateCallback).toHaveBeenLastCalledWith("my-value");
// //   });
// // });

// // export default Test;
