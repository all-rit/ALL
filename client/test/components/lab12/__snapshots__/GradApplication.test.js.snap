// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GradApplication Component renders correctly 1`] = `
<div>
  <div
    style={
      {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <div
      className="tw-mt-10 tw-shadow-2xl-top-bottom tw-rounded-3xl tw-min-w-72 tw-w-9/12 lg:tw-w-7/12 tw-h-full tw-p-10"
    >
      <h1
        className="tw-text-3xl tw-font-bold tw-mb-8"
      >
        Apply for Graduation at ALL University
      </h1>
      <div
        className="tw-text-left tw-p-6"
      >
        <h2
          className="tw-text-xl tw-font-semibold tw-mb-1.5"
        >
          Personal Information:
           
        </h2>
        <div
          className="sm:tw-flex tw-mb-6"
        >
          <label
            className="tw-pr-8"
          >
            Legal First Name:
          </label>
          <div
            className="tw-flex tw-flex-col tw-max-w-72 tw-h-8 tw-w-full sm:tw-w-8/12 md:tw-w-6/12"
          >
            <input
              onChange={[Function]}
              placeholder="Ex: Jane"
            />
          </div>
        </div>
        <div
          className="sm:tw-flex tw-mb-6"
        >
          <label
            className="tw-pr-8"
          >
            Legal Last Name:
          </label>
          <div
            className="tw-flex tw-flex-col tw-max-w-72 tw-h-8 tw-w-full sm:tw-w-8/12 md:tw-w-6/12"
          >
            <input
              onChange={[Function]}
              placeholder="Ex: Smith"
            />
          </div>
        </div>
        <div
          className="sm:tw-flex tw-mb-6"
        >
          <label
            className="tw-pr-8"
          >
            Pronouns:
          </label>
          <div
            className="tw-flex tw-flex-col tw-max-w-72 tw-h-8 tw-w-full sm:tw-w-8/12 md:tw-w-6/12"
          >
            <input
              onChange={[Function]}
              placeholder="Ex: They/Them"
            />
          </div>
        </div>
        <h3
          className="tw-mt-10 tw-text-xl tw-font-semibold tw-mb-1.5"
        >
          Academic Information:
           
        </h3>
        <div
          className="sm:tw-flex tw-mb-6"
        >
          <label
            className="tw-pr-8"
          >
            College:
          </label>
          <div
            className="tw-flex tw-flex-col tw-max-w-72 tw-h-8 tw-w-full sm:tw-w-8/12 md:tw-w-6/12"
          >
            <input
              onChange={[Function]}
              placeholder="Ex: RIT"
            />
          </div>
        </div>
        <div
          className="sm:tw-flex tw-mb-6"
        >
          <label
            className="tw-pr-8"
          >
            Major:
          </label>
          <div
            className="tw-flex tw-flex-col tw-max-w-72 tw-h-8 tw-w-full sm:tw-w-8/12 md:tw-w-6/12"
          >
            <input
              onChange={[Function]}
              placeholder="Ex: CS"
            />
            <div
              className="tw-mb-4"
            />
          </div>
        </div>
        <div
          className="sm:tw-flex tw-mb-6"
        >
          <label
            className="tw-pr-8"
          >
            Graduation Term:
          </label>
          <div
            className="tw-flex tw-flex-col tw-max-w-72 tw-h-8 tw-w-full sm:tw-w-8/12 md:tw-w-6/12"
          >
            <input
              onChange={[Function]}
              placeholder="Ex: Spring 2024"
            />
            <div
              className="tw-mb-4"
            />
          </div>
        </div>
      </div>
      <button
        className="tw-text-error-red tw-mt-8 btn-primary btn btn-md"
        onClick={[Function]}
      >
        Submit Application
      </button>
       
    </div>
  </div>
</div>
`;
